### PR TITLE
SD-1264 - Adds delivery history section to dedicated IPs

### PR DESCRIPTION
**Pull request by nicklemmon**
[SD-1264](https://jira.int.messagesystems.com/browse/SD-1264)

### What Changed

- Refactored `EditIpPage.js` from a class component to a functional component
- Refactored `EditIpPage.test.js` from Enzyme to React Testing Libary
- Added mock `IpForm` component to prevent React Testing Library tests from breaking due to presence of Redux Form
- Added new 10 day history chart for IP warmup
- Adds new `<Abbreviation/>` component, supporting tests, and supporting documentation to standardize styling and provide type checks for abbreviated content in the app
- Replaced instances of `<abbr>` with the `<Abbreviation/>` component - **NOTE:** This resulted in Prettier reformatting of several files that don't initially appear to be related to the task at hand
- Removed styles for `abbr[title]`
- Added handling for `.toHaveStyle` matcher that is common to both React Testing Library and Enzyme
- Added new Sass mixin that can be used to add re-usable line-chart styles to a component

#### The Feature
1. Update your `sparkpost.conf` to point to production and log in with a `108` account
1. Go to `/account/ip-pools` and click on an IP Pool with dedicated IPs assigned
1. Click on a dedicated IP
1. Verify the loading state in the panel at the bottom
1. Verify that at the bottom of the page, a table renders with a chart showing the last 10 days of sending
1. Inspect the DOM and verify that a visually hidden `<table>` exists that mirrors the data in the chart

#### Other Changes
- Verify instances of the `<abbr/>` tag were replaced with `<Abbreviation/>`
- Verify the docs don't have any typos and make sense to you. Run `npm run storybook-dev` and go to http://localhost:9001/?path=/docs/accessibility-abbreviation--page
- Open up `Abbreviations.test.js` and add the following. Verify that tests continue to pass (to test both Enzyme and React Testing Library matchers are working):
```js
it('renders enzyme styles', () => {
  const wrapper = mount(
    <Abbreviation title="Assistant Vice President" style={{ display: 'none' }}>
      AVP
    </Abbreviation>,
  );

  expect(wrapper.find('abbr')).toHaveStyle({ display: 'none' });
  expect(wrapper.find('abbr')).toHaveStyle('display', 'none');
});
```
- Go to `/reports/summary` and verify styles for the line chart match prod

### To Do
- [ ] Incorporate feedback

### CODE REVIEW

File: src/pages/ipPools/components/DeliveryHistoryLineChart.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+              tickFormatter={xTickFormatter}
+            >
+              <Label value="Date" offset={15} position="insideBottom" opacity={0.5} fontSize={16} />
+            </XAxis>
+
+            <Tooltip labelFormatter={tooltipLabelFormatter} />
+
+            <Line
+              type="monotone"
+              stroke="#2693C3"
```
Comment (shrirupa) [inline src/pages/ipPools/components/DeliveryHistoryLineChart.js]: We should capitalize _d_ in Deliveries for Tooltip. I think just add name="Deliveries" to <Line> and that should do it.

File: src/pages/ipPools/EditIpPage.js [hunk 2]
```diff
@@ -1,18 +1,40 @@
...
+    pageLoading,
+    pool,
+    pageError,
+    chartLoading,
+    chartError,
+    deliveryHistory,
+  } = props;
+
+  const handleSubmit = values => {
+    const { ip, history, showAlert } = props;
```
Comment (shrirupa) [inline src/pages/ipPools/EditIpPage.js]: not an important change, ip can be removed from here, it is already destructured outside.

File: src/components/abbreviation/tests/Abbreviation.test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    expect(queryByText('CSM')).toContainElement(document.querySelector('abbr'));
+    expect(queryByText('CSM')).toHaveAttribute('title', 'Customer Success Manager');
+  });
+
+  it('renders with the value passed in via the "className" prop along with the default "Abbreviation" class', () => {
+    const { queryByText } = render(
+      <Abbreviation title="Vice President" className="my-class">
+        VP
+      </Abbreviation>,
```
Comment (shrirupa) [inline src/components/abbreviation/tests/Abbreviation.test.js]:
```
it('renders enzyme styles', () => {
  const wrapper = mount(
    <Abbreviation title="Assistant Vice President" style={{ display: 'none' }}>
      AVP
    </Abbreviation>,
  );

  expect(wrapper.find('abbr')).toHaveStyle({ display: 'block' });
  expect(wrapper.find('abbr')).toHaveStyle('display', 'block');
});
```

appended the test file with this but the test didn't pass. It gave the following error

Expected <abbr> component style values to match for key "display", but they didn't

      48 |     );
      49 |   
    > 50 |     expect(wrapper.find('abbr')).toHaveStyle({ display: 'block' });
         |                                  ^
      51 |     expect(wrapper.find('abbr')).toHaveStyle('display', 'block');
      52 |   });
      53 | });

File: src/pages/ipPools/components/DeliveryHistoryLineChart.js (hunk 1 above)
Comment (nicklemmon) [reply src/pages/ipPools/components/DeliveryHistoryLineChart.js]: Good call. Got it working!

**Comment by shrirupa** [comment]
Looks good to me 👍

**Pull request merged**